### ISSUE TITLE
Different notification icons on per account basis

### ISSUE THREAD

**Issue by woj-tek** [k9mail/k-9#2239]
### Expected behavior
It would be nice if it would be possible to configure accounts to have different icons in the notifications

### Actual behavior
All notifications from all accounts have same icon.

### Steps to reproduce

Motivation: currently I'm using k9mail for my "work email" (because it handles imap better and I had some weird issue ages ago with gmail handling) and I'm using +typeApp for my "personal" emails (on gmail). The other day I was annoyed with the +type app not respecting synchronisation (and their response that "we know better and we will ignore system synchronisation setting") that I decided to switch to k9mail all my accounts and the experience was very positive (archiving works, yay). However I've noticed that I grew accustomed to being able to differentiate with a quick glance to which account email from particular notification arrived (I'm usually on silent mode so different sounds won't work).

I'm not sure if it's technically possible, but would be great!

### Environment
K-9 Mail version: 5.203

Android version: 6.0.1

Account type (IMAP, POP3, WebDAV/Exchange): n/a

**Comment by ployt0** [comment]
Can you link/show the icons +type uses? I'm happy with colors in the notification draw as I have many email accounts of the same type, personal.

**Comment by woj-tek** [comment]
OK, so I've grabbed a couple of screenshot (let's hope they are clear enough without arrows and borders):
* the difference between personal/work is easily noticeable with different icon (two on the left):
![notification1](http://i.imgur.com/YaeXJ9q.png)
* two accounts in k9mail share same icon (this actually applies to type-mail as well - multiple accounts will have same icons...)
![notification2](http://i.imgur.com/9oyp8m4.png)
* (cool addition) if there are multiple notifications the icon changes to "multiply items" (and this gave me idea that it should be possible to differentiate the icons based on conditions, so in this case - accounts):
![notification3](http://i.imgur.com/dgzKtwa.png)
* (another possibly cool feature related to notification icon) when there is only a single-item notification the icon in the opened notification changes to the user avatar:
![notification4](http://i.imgur.com/gfhgmDx.png)
![notification5](http://i.imgur.com/FwKqeZ3.png)

**Comment by Valodim** [comment]
An account setting to choose from a predefined list of icons should be relatively simple to do.

@cketti think this could work as a beginner issue?

**Comment by cketti** [comment]
I guess the code part is not too complicated and could be done by a beginner. But what would the different icons look like?
@k9mail/design-team: Any suggestions?

@woj-tek: For the contact picture feature, please create a new issue.

**Comment by Valodim** [comment]
Hmm, maybe a set of emojis?

**Comment by woj-tek** [comment]
@cketti I've created new issue as instructed.

As for the icons ( @Valodim ) I'm not sure emojis are suitable - wouldn't a set of a 5-ish (or a bit more) generic e-mail related icons be better idea? With the possibility for the user to use some custom icon? (I assume majority of the users have less than 5 mailboxes, and I would think that chances of having more than 10 is really, really slim)

**Comment by Valodim** [comment]
Makes sense. I suppose we can always decide on this set of five at a later point.

**Comment by juliakorbut** [comment]
I can totally help out. Ideally the default should already be pretty decent, users shouldn't have to go to settings to get legible notifications. 

What are our options aside from fixed icons? Can we use initials or letters? For example, display "K" for an account called "jk9@k9mail.com" etc?

Or can we go away from multiple status bar icons altogether? And just use one icon in the status bar for K-9, and then the notification itself would be split to multiple accounts?

**Comment by juliakorbut** [comment]
I can totally help out. Ideally the default should already be pretty decent, users shouldn't have to go to settings to get legible notifications. 

What are our options aside from fixed icons? Can we use initials or letters? For example, display "K" for an account called "k9@k9mail.com" etc?

Or can we go away from multiple status bar icons altogether? And just use one icon in the status bar for K-9, and then the notification itself would be split to multiple accounts?

**Comment by cketti** [comment]
Dynamically generated icons for notifications are supported on Android 6+. Status bar icons are monochrome.

I think most users welcome one notification icon per account. It's also what all other email clients are doing. I'd rather not move away from that.

More on notifications: https://material.io/guidelines/patterns/notifications.html

**Comment by woj-tek** [comment]
Initials are already used for contacts that doesn't have avatar so this may be confusing.

Also - I really do like separate icons for multiple accounts and this is actually the gist of the issue - to have and keep it, and to boot - make them easier to distinguish so I will know for which account mail arrived. Merging all accounts notification would simply, completely go against that.

**Comment by ployt0** [comment]
I like the idea but my notification bar tends quickly to get too crammed with icons I cannot distinguish that I won't be following up on this. _I_ _dont_ _receive_ _enough_ _unwanted_ _maill_
